File: samples/scoring/writer/handler.js
let MongoClient = require('mongodb').MongoClient;
let url = 'mongodb://mongo';
let getStdin = require("get-stdin");


getStdin().then(content => {
    console.log('received content');
    console.log(content);

    let request = JSON.parse(content);

    MongoClient.connect(url, function (err, client) {

        if (!err) {
            console.log('we are connected!');
        }

        const db = client.db('scoring');

        db.collection('users').insertOne(request);

        client.close();

    })
});

File: samples/scoring/scorer/handler.py
import pymongo
import sys

def get_stdin():
    buf = ""
    for line in sys.stdin:
        buf = buf + line
    return buf

name = get_stdin()

from pymongo import MongoClient
client = MongoClient('mongodb://mongo:27017')

db = client['scoring']

users = db.users

user = users.find_one({'name': name})

print('found user: ')
print(user)

user['score'] = 10

users.save(user)

print('saved user score successfully')

